### ISSUE TITLE
Make transport system pluggable

### ISSUE THREAD

**Issue by JamieCressey** [josegonzalez/python-beaver#379]
Moving the transports to a modular system have a number of advantages:

1) Users can install only the transports they require keeping the core code base lean.
2) Additional transports can be developed in an isolated environment, quickly and efficiently. 
3) Testing both the core code base and individual transports become a lot simpler in their own repos.

Thoughts?

**Comment by josegonzalez** [comment]
I *think* this is already possible as noted in #95, though maybe not well documented?

**Comment by JamieCressey** [comment]
Agreed, the majority of the works appears to have been completed in #95 (and certainly made it easier when I added a couple of transports previously), however there still needs to be a method of installing pluggable transports (if they're outside of the core repo), preferably via pip, and the existing ones need splitting out into their own repo.

**Comment by josegonzalez** [comment]
Should we make a python-beaver org and move the project there?

**Comment by josegonzalez** [comment]
Moved!

**Comment by JamieCressey** [comment]
I've made a start on the SQS module - https://github.com/python-beaver/beaver-sqs

I'll be testing it on a subset of our servers next week when I get back into the office. In the meantime I'll branch off and look at a more user friendly way of configuring the new modular layout.

**Comment by josegonzalez** [comment]
Nice nice. Let me know how else I can help. You/I may need to re-enable tests on travis now that it is in it's own organization.
